hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -1061,8 +1061,14 @@ public static SingularAssociationAttributeMapping buildSingularAssociationAttrib
 						sessionFactory
 				);
 
-		final FetchTiming fetchTiming = FetchStrategyHelper.determineFetchTiming( fetchStyle, type, sessionFactory );
+		final FetchTiming fetchTiming;
 
+		if ( !( (ToOne) bootProperty.getValue() ).isLazy() ) {
+			fetchTiming = FetchTiming.IMMEDIATE;
```

Comment:
There is a subtlety here. Previously when fetch mode is `JOIN`, IMMEDIATE is always returned; now if `FetchType.LAZY` is specified together with `FetchMode.JOIN`, IMMEDIATE won't be returned.

Reply:
That is true.  Each can affect the other.

Reply:
Oh, wait...

This is while determining the *mapped* fetching strategy.  I missed that part when I replied.  I *think* this is actually safe here.  @NathanQingyangXu do you have a specific case you are thinking of where this would happen?

@dreab8 Another consideration would be enhanced proxies.  I forget how we ended up implementing that wrt `#isLazy`.  Perhaps you could add a test for that as well.

Reply:
I realised the change I made cause for sure an issue in case of `OneToOne` when the value is nullable in sucha a case the `fetchTiming` should be Immediate no matter if the fetch type is set lazy.
Probably the code should be
```
if ( ( value instanceof OneToOne && value.isNullable() ) || !( value ).isLazy() ) {
			fetchTiming = FetchTiming.IMMEDIATE;
		}
...
```

Reply:
I am thinking of the following scenario:
```
@ManyToOne(fetch = FetchType.LAZY)
@Fetch(FetchMode.JOIN)
private SimpleEntity manyToOne;
```
The code change in this PR will treat it as 'lazy', but from user guide, the `join mode' will lead to 'eager'. I didn't explore it fully, but I am not sure this scenario will lead to some invariance violation or even runtime exception?

Reply:
however, it would be great to test the above scenario behaves the same between v5 and v6. Maybe I am wrong, but...

Reply:
I added a test also for this case, basically I run all the tests on v5 and then ported them on v6